### PR TITLE
Add Bernoulli Sampling LOp

### PR DESCRIPTION
**Pull request by lorenzhs**
This is a work in progress

### CODE REVIEW

File: thrill/api/sample.hpp [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    // Sampling rate
+    const double p_;
+    // Whether to generate skip values with a geometric distribution or to use
+    // the naive method
+    const bool use_skip_;
+    // Random generator
+    std::default_random_engine engine_ { std::random_device{}() };
+    std::bernoulli_distribution simple_dist_;
+    std::geometric_distribution<> skip_dist_;
+    typename decltype(skip_dist_)::result_type skip_remaining_ = -1;
```
Comment (bingmann) [inline thrill/api/sample.hpp]: All these decltype call for a using SkipDistribution = ...

File: tests/api/sampling_test.cpp [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+TEST(SamplingNode, CompileAndExecute) {
+
+    std::function<void(Context&)> start_func =
+        [](Context& ctx) {
+            size_t n = 1024;
+
+            auto sizets = Generate(
+                ctx,
+                [](const size_t& index) { return index; },
```
Comment (bingmann) [inline tests/api/sampling_test.cpp]: The identity lambda can be skipped. Generates integers by default

**Comment by lorenzhs** [comment]
Thanks, should be fixed

**Comment by lorenzhs** [comment]
Travis got stuck again: https://travis-ci.org/thrill/thrill/jobs/113070597

**Pull request merged**